components/map.jsx

```diff
@@ -70,6 +71,40 @@ var MarkerPopup = React.createClass({
   }
 });
 
+var ClubStatusLabel = React.createClass({
+  propTypes: {
+    status: React.PropTypes.string.isRequired,
+    showApproved: React.PropTypes.bool
+  },
+  render: function() {
+    if (this.props.status === 'pending') {
+      return (
+        <span className="label label-warning"
+         title="This club is pending approval and is not visible to other users.">
+          pending
+        </span>
+      );
+    } else if (this.props.status === 'denied') {
+      return (
+        <span className="label label-danger"
+         title="This club has been denied approval and is not visible to other users.">
+          denied
+        </span>
+      );
+    }
+    if (this.props.showApproved) {
```

Comment:
That's a good point, though I also want to make sure the code is fault-tolerant so that e.g. the renderer doesn't crash if things change....

Oooo, I know!! This looks like a good situation for [`React.PropTypes.oneOf`](https://facebook.github.io/react/docs/reusable-components.html#prop-validation)!! That way if our assumptions are ever incorrect, we'll get a console message logged in dev mode, but things won't crash in production.